java/lib/src/test/java/truelayer/signing/UsageTest.java

```diff
@@ -0,0 +1,311 @@
+package truelayer.signing;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import static java.nio.file.Files.readAllBytes;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+public class UsageTest {
+
+    static String kid = "45fc75cf-5649-4134-84b3-192c2c78e990";
+
+    @Test
+    public void fullSignature() throws IOException {
+
+        byte[] privateKey = readAllBytes(new File("src/test/resources/ec512-private.pem").toPath());
+        byte[] publicKey = readAllBytes(new File("src/test/resources/ec512-public.pem").toPath());
+
+        byte[] body = "{\"currency\":\"GBP\",\"max_amount_in_minor\":5000000}".getBytes(StandardCharsets.UTF_8);
+        String idempotencyKey = "idemp-2076717c-9005-4811-a321-9e0787fa0382";
+        String path = "/merchant_accounts/a61acaef-ee05-4077-92f3-25543a11bd8d/sweeping";
+
+        String tlSignature = Signer.from(kid, privateKey)
+                .header("Idempotency-Key", idempotencyKey)
+                .method("post")
+                .path(path)
+                .body(body)
+                .sign();
+
+        Verifier.from(publicKey)
+                .method("POST")
+                .path(path)
+                .header("X-Whatever", "aoitbeh")
+                .header("Idempotency-Key", idempotencyKey)
+                .body(body)
+                .requiredHeader("Idempotency-Key")
+                .verify(tlSignature); // should not throw
+    }
+
+    @Test
+    public void verifyBodyStaticSignature() throws IOException {
```

Comment:
The "body static" probably comes from the rust lib that supports body-only signing too.

```suggestion
    public void verifyStaticSignature() throws IOException {
```